crates/bevy_pbr/src/render/mesh.rs

```diff
@@ -734,6 +768,14 @@ impl SpecializedMeshPipeline for MeshPipeline {
             vertex_attributes.push(Mesh::ATTRIBUTE_COLOR.at_shader_location(4));
         }
 
+        if cfg!(any(not(feature = "webgl"), not(target_arch = "wasm32"))) {
+            shader_defs.push("PREPASS_DEPTH_SUPPORTED".into());
```

Comment:
We already have the `DEPTH_PREPASS` shader define, and it is not clear to me why this is needed on top of that, as shader code would end up being gated behind both in most cases.

If the purpose is to gate prepass related shader code when the platform does not support it, surely a better solution would be to prevent the depth prepass from being enabled in the first place? 

This would avoid clutter in shader code that would otherwise have to check for both macros (`#ifdef DEPTH_PREPASS` immediately followed by `#ifdef PREPASS_DEPTH_SUPPORTED`) and avoid whatever confusing behaviour results from returning 0 for depth whenever the prepass is not supported (elsewhere in this PR).

Reply:
> We already have the DEPTH_PREPASS shader define, and it is not clear to me why this is needed on top of that, as shader code would end up being gated behind both in most cases.

The existing `DEPTH_PREPASS` shader def is set based on the `MeshPipelineKey::DEPTH_PREPASS` pipeline key, which (if I understand it correctly) is only enabled from [`queue_prepass_material_meshes()`](https://github.com/bevyengine/bevy/blob/c353b41bd64695d7213bae8924b871f9c428f580/crates/bevy_pbr/src/prepass/mod.rs#L787) when we're _currently_ performing the depth prepass, _not_ for the main pass _if_ the prepass is enabled. (Unlike other keys like, say,`MeshPipelineKey::TAA`)

> If the purpose is to gate prepass related shader code when the platform does not support it, surely a better solution would be to prevent the depth prepass from being enabled in the first place?

That's a valid and probably cleaner solution, however I'm not sure how to get that information [from here](https://github.com/bevyengine/bevy/blob/592a6c921aa06d3217520bdf1933b875e3aba192/crates/bevy_pbr/src/render/mesh.rs#L772) without introducing a second `MeshPipelineKey::DEPTH_PREPASS_ENABLED` key. (Perhaps there's a better place to set this shader def instead?)

> This would avoid clutter in shader code that would otherwise have to check for both macros (#ifdef DEPTH_PREPASS immediately followed by #ifdef PREPASS_DEPTH_SUPPORTED)

Not sure if a typo, but just to make sure we're on the same page, that first check is an `#ifndef` (not an `#ifdef`) which is maybe why the second flag feels redundant?

> and avoid whatever confusing behaviour results from returning 0 for depth whenever the prepass is not supported (elsewhere in this PR).

That's only on the `show_prepass` shader, so that we show a black screen instead of crashing. (I can remove it if crashing is more explicit for that demo when depth prepass is unsupported)